modules/sor/sorV2/sorV2.service.ts

```diff
@@ -0,0 +1,243 @@
+import { 
+    BasePool, 
+    ChainId,  
+    sorGetSwapsWithPools, 
+    Token, 
+    Address, 
+    SwapKind, 
+    sorParseRawPools, 
+    RawStablePool, 
+    RawLinearPool, 
+    RawWeightedPool, 
+    RawComposableStablePool, 
+    RawMetaStablePool,
+    Swap
+} from '@balancer/sdk';
+import { GqlSorGetSwapsResponseNew, GqlSorSwapType } from '../../../schema';
+import { PrismaPoolType, PrismaToken } from '@prisma/client';
+import { GetSwapsInput } from '../sor.service';
+import { tokenService } from '../../token/token.service';
+import { networkContext } from '../../network/network-context.service';
+import { prisma } from '../../../prisma/prisma-client';
+import { PrismaPoolWithDynamic, prismaPoolWithDynamic } from '../../../prisma/prisma-types';
+import { RawPool } from '@balancer/sdk';
+import { HumanAmount, SupportedRawPoolTypes } from '@balancer/sdk';
+
+import { getSwapCompare } from './temp';
+
+// TODO - Check if this has been deployed to same address across networks?
+export const SOR_QUERIES = '0x1814a3b3e4362caf4eb54cd85b82d39bd7b34e41';
+
+export class SorV2Service {
+    public async getSwaps({
+        tokenIn,
+        tokenOut,
+        swapType,
+        swapAmount,
+    }: GetSwapsInput): Promise<GqlSorGetSwapsResponseNew> {
+        // TODO - This takes ~1.262s on my local machine. Is this fast enough? Any obvious ways to improve?
+        console.time('poolsFromDb');
+        const poolsFromDb = await this.getBasePoolsFromDb();
+        console.timeEnd('poolsFromDb');
+        const chainId = networkContext.chainId as unknown as ChainId;
+        const tIn = await this.getToken(tokenIn as Address, chainId);
+        const tOut = await this.getToken(tokenOut as Address, chainId);
+        const swap = await sorGetSwapsWithPools(
+                    tIn,
+                    tOut,
+                    this.mapSwapType(swapType),
+                    swapAmount,
+                    poolsFromDb,
+                    // swapOptions, // TODO - Handle properly
```

Comment:
We would also want to add default SOR swapOptions into the config, could be over written by passing to the resolver. Take a look at our lambda implementation https://github.com/beethovenxfi/sor-lambda/blob/main/src-ts/sorGetSwaps.ts#L51-L57